plugins/gnmi_nvos_events_plugin/gnmi_events_server/gnmi_events_receiver.py

```diff
@@ -0,0 +1,116 @@
+        self.traps_number = 0
+        self.throttling_interval = 10
+        self.high_event_rate = 100
+        self.throttling_thread = None
+        self.events = []
+
+    def manager(self):
+        for ip, switch in self.switch_dict.items():
+            if not switch.socket_thread:
+                switch.socket_thread = threading.Thread(target=self.subscribe,
+                                                        args=(ip,
+                                                              switch.user,
+                                                              switch.credentials,
+                                                              switch.guid))
+                switch.socket_thread.start()
+        if not self.throttling_thread:
+            self.throttling_thread = threading.Thread(target=self.throttle_events)
+            self.throttling_thread.start()
+
+    def cleanup(self):
+        for switch in self.switch_dict.values():
+            if switch.socket_thread:
+                switch.socket_thread.join(timeout=self.throttling_interval)
+
+    def subscribe(self, ip, user, credentials, guid):
+        try:
+            with gNMIclient(target=(ip, helpers.ConfigParser.gnmi_port),
+                                    username=user,
+                                    password=credentials,
+                                    skip_verify=True) as gc:
+                events_stream = gc.subscribe_stream(subscribe=self.SUBSCRIBE_CONF, target=self.TARGET)
+                init = True
+                for event_dict in events_stream:
+                    if init:
+                        init = False
+                        continue
+                    # print(json.dumps(event_dict, indent=4))
+                    # print("ENTRY END")
+                    self.traps_number += 1
+                    payload = {"object_name": guid, "otype": "Switch"}
+                    event_update = event_dict.get("update")
+                    if event_update:
+                        event_update = event_update.get("update")
+                        description = ""
+                        resource = ""
+                        for event in event_update:
+                            path = event.get("path")
+                            if path == "state/text":
+                                description = event.get("val")
+                            elif path == "state/resource":
+                                resource = event.get("val")
+                            elif path == "state/severity":
+                                payload["event_id"] = helpers.Severity.LEVEL_TO_EVENT_ID.get(event.get("val"))
+                        payload["description"] = f"{resource}: {description}" if resource else description
+                        self.events.append(payload)
+        except Exception as e:
+            logging.error("Failed to create gNMI socket for %s: %s", ip, e)
+
+    def throttle_events(self):
+        while True:
```

Comment:
why to have this thread busy waiting for events? (Most of the time this thread should be idle and wait for events from all the switches...) either add a configurable sleep between the check or use a queue that this thread is listening to when new events are coming from any of the switches...